modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountHelperImpl.java

```diff
@@ -374,6 +374,20 @@ public AccountEntry getCurrentAccountEntry(
 				return accountEntry;
 			}
 
+			PermissionChecker permissionChecker =
+				PermissionThreadLocal.getPermissionChecker();
+
+			if (permissionChecker.hasPermission(
+					commerceChannelGroupId, AccountEntry.class.getName(),
+					commerceChannel.getCompanyId(), ActionKeys.VIEW)) {
+
+				setCurrentCommerceAccount(
+					httpServletRequest, commerceChannelGroupId,
+					accountEntry.getAccountEntryId());
+
```

Comment:
The check right after this is:

```
		for (AccountEntry commerceChannelAccountEntry :
					commerceChannelAccountEntries) {
				if (accountEntry.getAccountEntryId() ==
						commerceChannelAccountEntry.getAccountEntryId()) {
					setCurrentCommerceAccount(
						httpServletRequest, commerceChannelGroupId,
						accountEntry.getAccountEntryId());
					return accountEntry;
				}
			}
```
Without a check like this (you would have to avoid the loop since that is the entire goal of this PR), there will be a possibility that the accountEntry that you are setting is not availaboe to the channel.

Reply:
@brianikim I've added the count == 0 condition to address this.

I've taken it from the [filtering](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountHelperImpl.java#L527-L557) done when looping through the accounts.
If the count of CommerceChannelAccountEntryRels is 0, that means the channel doesn't have any account filter.
If the filter is activated and the selected account is available in the channel, it will be fetched in the [previous checks](https://github.com/liferay/liferay-portal/blob/master/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/util/CommerceAccountHelperImpl.java#L346-L375), and if there is no Rel and the count isn't 0, that means that the channel has an account filter enabled and the selected account is not included in it.

I've also confirmed with Stefano that this doesn't require a test since it's just a refactoring for performance

Reply:
@victorcdp , looks fine with me. Go ahead with further review request from Stefano.